lib/logstash/filters/grok.rb

```diff
@@ -288,21 +288,28 @@ def filter(event)
         end
         #break if done
       end # @patterns.each
-  
+
       if matched
         filter_matched(event)
       else
         # Tag this event if we can't parse it. We can use this later to
         # reparse+reindex logs if we improve the patterns given.
         @tag_on_failure.each do |tag|
-          event["tags"] ||= []
-          event["tags"] << tag unless event["tags"].include?(tag)
+          # do not replace the code below with:
+          #   event["tags"] ||= []
+          #   event["tags"] << tag unless event["tags"].include?(tag)
+          # this assumes implementation specific feature of returning a mutable object
+          # from a field ref which should not be assumed and will change in the future.
+
+          tags = event["tags"] || []
```

Comment:
oh yes, good catch, I don't know why I did not correctly clean this. is can be replaced with a simple: 
```ruby
event.tag(tag)
```